File: src/main/java/org/shark/example/asyncTask/AsyncTask2.java
package org.shark.example.asyncTask;

import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.concurrent.CompletableFuture;

@Service
@Slf4j
public class AsyncTask2 {

    @Async("taskExecutor2")
    public CompletableFuture<String> start(String value1) {
        String time = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
        log.info("time = " + time);
        try {
            Thread.sleep(3000);
        } catch (InterruptedException e) {
            throw new RuntimeException(e);
        }
        return CompletableFuture.completedFuture("AsyncTask1 value = " + value1);
    }
}

File: src/main/kotlin/org/shark/example/service/base/async/AsyncTask2.kt
package org.shark.example.service.base.async

import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Service
import java.text.SimpleDateFormat
import java.util.*
import java.util.concurrent.CompletableFuture

@Service
class AsyncTask2 {

    private val log = LoggerFactory.getLogger(AsyncTask2::class.java)

    @Async("taskExecutor2")
    fun start(value1: String): CompletableFuture<String> {
        val time = SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(Date())
        log.info("time = {}", time)
        try {
            Thread.sleep(3000)
        } catch (e: InterruptedException) {
            throw RuntimeException(e)
        }
        return CompletableFuture.completedFuture("AsyncTask1 value = $value1")
    }
}
